src/components/search/Search.svelte

```diff
@@ -1,41 +1,64 @@
 <script>
-  import { tick } from 'svelte';
+import { tick } from 'svelte';
 import { fly } from 'svelte/transition';
+import { cubicOut } from 'svelte/easing';
 import SearchIcon from 'udgl/icons/Search.svelte';
 import LineSegSpinner from 'udgl/LineSegSpinner.svelte';
+import { throttle } from 'throttle-debounce';
 
-import {
-    store,
-} from '../../state/store';
-
-
-import telemetrySearch from '../../state/telemetry-search';
-
+import { getSearchResults } from '../../state/api';
 import TelemetrySearchResults from './SearchResults.svelte';
 
 
 let inputElement;
 let searchContainer;
+let results = [];
+let searchIsActive = false;
+let searchQuery = '';
+
+const SEARCH_THROTTLE_TIME = 500; // how often to send the PSS fetch (in ms)
+const SEARCH_RESULTS_LIMIT = 15; // maximum number of results to show
 
 function turnOnSearch() {
-    store.setField('searchIsActive', true);
+  searchIsActive = true;
 }
 
 function unfocus() {
-    inputElement.blur();
-    store.setField('searchIsActive', false);
+  inputElement.blur();
+  searchIsActive = false;
 }
 
 async function onKeypress(event) {
-    if ($store.searchIsActive) {
-      const { key } = event;
-      if (key === 'Escape') {
-        await tick();
-        unfocus();
-      }
+  if (searchIsActive) {
+    const { key } = event;
+    if (key === 'Escape') {
+      await tick();
+      unfocus();
     }
+  }
 }
 
+let searchResults = Promise.resolve([]);
+let query = '';
+
+const handleSearchInput = throttle(SEARCH_THROTTLE_TIME, ({target: {value}}) => {
+  query = value;
+  searchResults = getSearchResults(query, SEARCH_RESULTS_LIMIT).then((r) => {
+    // sort these?
+    if (r.constructor === Array) {
+      results = r.sort((a, b) => {
+        const aHas = a.name.toLowerCase().includes(query);
+        const bHas = b.name.toLowerCase().includes(query);
+        if (aHas && !bHas) return -1;
+        if (bHas && !aHas) return 1;
+        return 0;
+      });
```

Comment:
In the near future I plan to update the probe-search to rank the results based on weighting of the matches. Once that lands this could likely go away. Just a heads up.

Reply:
Yea I'd prefer if this were gone.